docs/design/operator.md

```diff
@@ -0,0 +1,407 @@
+            nodeSelectorTerms:
+            - matchExpressions:
+              - key: storage
+                operator: In
+                values:
+                - node
+      tolerations:
+      - key: storage
+        operator: Exists
+      resources:
+        registrar:
+          limits:
+            cpu: '200'
+            memory: '500'
+          requests:
+            cpu: '100'
+            memory: '250'
+        liveness:
+          limits:
+            cpu: '200'
+            memory: '500'
+          requests:
+            cpu: '100'
+            memory: '250'
+        plugin:
+          limits:
+            cpu: '200'
+            memory: '500'
+          requests:
+            cpu: '100'
+            memory: '250'
+      volumes:
+      - name: host-run
+        volumes:
+          hostPath:
+            path: "/run"
+            type: Directory
+        volumeMounts:
+          name: ''
+          readOnly: true
+          mountPath: "/run"
+          mountPropagation: Bidirectional
+      kubeletDirPath: "/var/lib/kubelet"
+      imagePullPolicy: IfNotPresent
+    provisioner:
+      priorityClassName: system-cluster-critical
+      labels:
+        app: provisioner
+      annotations:
+        k8s.v1.cni.cncf.io/networks: macvlan-conf-1
+      provisionerReplicas: 2
+      affinity:
+        nodeAffinity:
+          requiredDuringSchedulingIgnoredDuringExecution:
+            nodeSelectorTerms:
+            - matchExpressions:
+              - key: storage
+                operator: In
+                values:
+                - node
+      tolerations:
+      - key: storage
+        operator: Exists
+      resources:
+        attacher:
+          limits:
+            cpu: '200'
+            memory: '500'
+          requests:
+            cpu: '100'
+            memory: '250'
+        snapshotter:
+          limits:
+            cpu: '200'
+            memory: '500'
+          requests:
+            cpu: '100'
+            memory: '250'
+        resizer:
+          limits:
+            cpu: '200'
+            memory: '500'
+          requests:
+            cpu: '100'
+            memory: '250'
+        provisioner:
+          limits:
+            cpu: '200'
+            memory: '500'
+          requests:
+            cpu: '100'
+            memory: '250'
+        omapGenerator:
+          limits:
+            cpu: '200'
+            memory: '500'
+          requests:
+            cpu: '100'
+            memory: '250'
+        liveness:
+          limits:
+            cpu: '200'
+            memory: '500'
+          requests:
+            cpu: '100'
+            memory: '250'
+        plugin:
+          limits:
+            cpu: '200'
+            memory: '500'
+          requests:
+            cpu: '100'
+            memory: '250'
+    liveness:
+      metricsPort: 8000
+    leaderElection:
+      leaseDuration: 100
+      renewDeadline: 100
+      retryPeriod: 10
+    deployCSIAddons: true
+    cephFsKernelClient: true
```

Comment:
Shouldn't cephfs settings only be on a cephfs-specific setting? I thought the settings in this CRD should apply to all drivers.

Reply:
we have two options either we can have sections for each driver or we don't need extra section we can go with driver_type_configuration like that. as we will have very minimal configuration and sometimes a single configuration can be applicable for 2 drivers as well, for that reason we kept this way.

Reply:
This is specific to multiple cephfs driver deployed by the single/multiple cephcsi operator, as this is operator config its global configuration for all the cephfs drivers its managing.

Reply:
@travisn this CR offers a place to overwrite all default values, even if only applied to a single driver type. 
This is not a configuration for a driver and will not result in deployment of a driver

Reply:
Ok understood now on why these settings are needed in this CRD. What about a section for the `cephfs` type, just to organize the settings? 
```suggestion
    cephfs:
      kernelClient: true
      kernelMountOptions: ms_mode=secure
```

Reply:
We deliberated toward and away from a dedicated section. There are good reasons to do it and good reasons to not. 

But consider the following: For a single driver, a section does not make sense as it can only be of one type - while a section suggests that you should be able to specify many.

Given that the above was the leading reason we opted out of a section for the driver CR. We decided to match the operator's config CR with an identical structure, to simplify the coding (reuse the same struct) and for familiarity and consistency.

Reply:
@travisn if you are still not okay with this, am tracking similar one here https://github.com/ceph/ceph-csi-operator/issues/6 we can discuss on that

Reply:
It's a small detail, it's fine either way